pkg/operator/ceph/cluster/controller.go

```diff
@@ -398,6 +398,12 @@ func (c *ClusterController) requestClusterDelete(cluster *cephv1.CephCluster) (r
 		return reconcile.Result{}, true
 	}
 
+	// If encrypted with KMS let's remove keys?
+	// Not sure since typically we don't remove the PVC so we might still want to have the keys to recover the volumes
+	if cluster.Spec.Storage.IsOnPVCEncrypted() && cluster.Spec.Security.KeyManagementService.IsEnabled() {
+		// remove keys for each PV!
```

Comment:
TODO?

Reply:
This is something we need to discuss. I'm a bit hesitant since the default policy is not to delete PVC once we delete the cluster. If that's the case, the keys must remain. I need to dig into this a little bit.